streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -244,24 +244,43 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
     }
 
     if (isTrackerStopping || isTrackerStopped) {
-      false
-    } else if (!scheduleReceiver(streamId).contains(hostPort)) {
-      // Refuse it since it's scheduled to a wrong executor
-      false
+      return false
+    }
+
+    val scheduledExecutors = receiverTrackingInfos(streamId).scheduledExecutors
+    if (scheduledExecutors.nonEmpty) {
+      // This receiver is registering at the first time, it's scheduled by
+      // ReceiverSchedulingPolicy.scheduleReceivers. So use "scheduledExecutors" to check it.
+      if (!scheduledExecutors.get.contains(hostPort)) {
```

Comment:
Same question as below, why handle the scheduling and rescheduling cases differently.